packages/opentelemetry-exporter-collector/src/CollectorExporterBase.ts

```diff
@@ -132,14 +143,106 @@ export abstract class CollectorExporterBase<
     return this._shuttingDownPromise;
   }
 
+  private _getUrl(config: T): string {
```

Comment:
The cyclomatic complexity of this function is really high 16-17 if I counted correctly.
The whole purpose of splitting the classes was to put the final information in the final class like for example name, default url etc.. Imagine now you have 50 different exporters and you want to create a default url for them. I think doing this in base class becomes very complicated as it will produce an enormous complex function, not even mentioning about all possible tests cases.